module/state_synchronization/requester/oneshot_execution_data_requester_test.go

```diff
@@ -0,0 +1,127 @@
+package requester
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/module/blobs"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data/cache"
+	edmock "github.com/onflow/flow-go/module/executiondatasync/execution_data/mock"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/mempool/herocache"
+	"github.com/onflow/flow-go/module/metrics"
+	synctest "github.com/onflow/flow-go/module/state_synchronization/requester/unittest"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type OneshotExecutionDataRequesterSuite struct {
+	suite.Suite
+}
+
+func TestRawExecutionDataRequesterSuite(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(OneshotExecutionDataRequesterSuite))
+}
+
+func (suite *OneshotExecutionDataRequesterSuite) TestRequester_RawRequestExecutionData() {
+	logger := unittest.Logger()
+	metricsCollector := metrics.NewNoopCollector()
+	config := OneshotExecutionDataConfig{
+		FetchTimeout:    DefaultFetchTimeout,
+		MaxFetchTimeout: DefaultMaxFetchTimeout,
+		RetryDelay:      DefaultRetryDelay,
+		MaxRetryDelay:   DefaultMaxRetryDelay,
+	}
+
+	headers := new(storagemock.Headers)
```

Comment:
use constructors for mocks so we get assertions automatically
```suggestion
	headers := storagemock.NewHeaders(suite.T())
```